tests/unit/models/resource/test_resource.py

```diff
@@ -70,3 +70,44 @@ def test_id_property_with_numeric_id():
 
     assert resource.id == "1024"
     assert isinstance(resource.id, str)
+
+
+def test_case_conversion():
+    resource_data = {"id": "abc-123", "FullName": "Alice Smith"}
+
+    resource = Model(resource_data)
+
+    assert resource.full_name == "Alice Smith"
+    assert resource.to_dict() == resource_data
+    with pytest.raises(AttributeError):
```

Comment:
Honestly speaking I would split it to two functions, just because you are checking here different things --> key-access and `to_dict` method